protocol/protocol.tex

```diff
@@ -5102,171 +5968,365 @@
+this allows also deriving the $\NullifierKey$ component of the \fullViewingKey. \nufive{(In
+\Orchard, that party needs the $\NullifierKey$ directly to make the \zkSNARKProof.)} Together
+$\AuthSignPublic$ and $\NullifierKey$ are sufficient to recognize spent \notes and to
+recognize and decrypt incoming \notes. However, the other party will not obtain spending
+authority for other \transactions, since it is not able to create a \spendAuthSignature by itself.
+} %pnote
+} %sapling
+
+
+\intropart
+\lsubsection{Note Commitments and Nullifiers}{commitmentsandnullifiers}
+
+A \transaction that contains one or more
+\joinSplitDescriptions\sapling{ or \spendDescriptions}, when entered
+into the \blockChain, appends to the \noteCommitmentTree with all constituent
+\noteCommitments.
+
+All of the constituent \defining{\nullifiers} are also entered into the
+\nullifierSet of the associated \treestate. A \transaction is not valid if it
+would have added a \nullifier to the \nullifierSet that already exists in the set
+(see \crossref{nullifierset}).
+
+\vspace{2ex}
+\sprout{Each}\notsprout{In \Sprout, each} \note has a $\NoteUniqueRand$ component.
+
+\sapling{
+\vspace{2ex}
+\introlist
+In \Sapling, each \defining{\positionedNote} has an associated $\NoteUniqueRand$ value which
+is computed from its \noteCommitment $\cm$ and \notePosition $\NotePosition$
+as follows:
+
+\begin{formulae}
+  \item $\NoteUniqueRand := \MixingPedersenHash(\cm, \NotePosition)$.
+\end{formulae}
+
+$\MixingPedersenHash$ is defined in \crossref{concretemixinghash}.
+} %sapling
 
 \vspace{2ex}
-Let $\PRFnf{}{}$\sapling{ and $\PRFnfSapling{}{}$} be as instantiated in \crossref{concreteprfs}.
+Let $\PRFnf{Sprout}{}$\sapling{ and $\PRFnf{Sapling}{}$}\nufive{ and $\PRFnf{Orchard}{}$}
+be as instantiated in \crossref{concreteprfs}.
 
 \vspace{2ex}
-\sprout{The \nullifier of a \note}\notsprout{For a \Sprout{} \note, the \nullifier}
-is derived as $\PRFnf{\AuthPrivate}(\NoteAddressRand)$, where $\AuthPrivate$ is the
-\spendingKey associated with the \note.
+For a \Sprout \note, the \nullifier is derived as $\PRFnf{Sprout}{\AuthPrivate}(\NoteUniqueRand)$,
+where $\AuthPrivate$ is the \spendingKey associated with the \note.
 
 \vspace{2ex}
 \sapling{
-For a \Sapling{} \note, the \nullifier is derived as
-$\PRFnfSapling{\AuthProvePublicRepr}(\NoteAddressRandRepr)$, where $\AuthProvePublicRepr$
-is a representation of the \nullifierDerivingKey associated with the \note and $\NoteAddressRandRepr = \reprJ(\NoteAddressRand)$.
+For a \Sapling \note, the \nullifier is derived as
+$\PRFnf{Sapling}{\NullifierKeyRepr}(\NoteUniqueRandRepr)$, where $\NullifierKeyRepr$
+is a representation of the \nullifierDerivingKey associated with the \note and
+$\NoteUniqueRandRepr = \reprJ(\NoteUniqueRand)$.
 } %sapling
 
+\vspace{2ex}
+\nufive{
+For an \Orchard \note, the \nullifier is derived as
+$\PRFnf{Orchard}{\NullifierKeyRepr}(\NoteUniqueRandRepr)$, where $\NullifierKeyRepr$
```

Comment:
This is currently copy-pasted from Sapling; it needs to be updated to use the nullifier scheme described in https://zcash.github.io/orchard/design/nullifiers.html